src/app/state/playback_state.rs

```diff
@@ -312,8 +313,11 @@ impl PlaybackState {
         let len = self.running_order().len();
         let next = self
             .position
-            .filter(|&p| p.index + 1 < len)
-            .map(|p| p.index + 1);
+            .filter(|&p| p.index + 1 < len || !matches!(self.repeat, RepeatMode::None))
```

Comment:
maybe and_then instead of filter+map? so that all three cases can be covered in the same closure

Reply:
Alright I will look into that. Thanks